src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ResettableServicePass.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Bundle\FrameworkBundle\EventListener\ServiceResetSubscriber;
+use Symfony\Component\DependencyInjection\Argument\IteratorArgument;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Alexander M. Turek <me@derrabus.de>
+ *
+ * @internal
+ */
+class ResettableServicePass implements CompilerPassInterface
+{
+    const RESET_TAG = 'kernel.reset';
+    const METHOD_ATTRIBUTE = 'method';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $services = $methods = array();
+
+        foreach ($container->findTaggedServiceIds(self::RESET_TAG, true) as $id => $tags) {
+            $services[$id] = new Reference($id, ContainerInterface::IGNORE_ON_UNINITIALIZED_REFERENCE);
+            $methods[$id] = array();
+
+            foreach ($tags as $attributes) {
+                if (!isset($attributes[self::METHOD_ATTRIBUTE])) {
+                    throw new \RuntimeException(
+                        sprintf('Tag %s requires the attribute %s to be set.', self::RESET_TAG, self::METHOD_ATTRIBUTE)
+                    );
+                }
+
+                $methods[$id][] = $attributes[self::METHOD_ATTRIBUTE];
```

Comment:
Do we really need to allow several methods? I'm not sure I see a use case where a class would require two calls to be resetted, do you?

Reply:
I don't, but I did not see a reason to forbid it. Also, I would need to limit the `kernel.reset` tag to max one tag per service.

Reply:
In such cases, we usually just ignore the next tags and always take index `0`.
We should do the same here IMHO.